### PR TITLE
[expo] Upgrade react-native-maps to 0.24.2

### PR DESCRIPTION
**Pull request by sjchmiela**
Just a regular dependency update. 🙂

### CODE REVIEW

**Review by mczernek — COMMENTED** [review]
Don't we want to use community version? Are you sure that using our fork is still needed?

**Comment by sjchmiela** [comment]
I updated our fork to match the community master, so now the only difference is:

https://github.com/react-native-community/react-native-maps/compare/master...expo:master

**Comment by mczernek** [comment]
Yeah, that's why i think we might migrate to community repo - it will require less work to update in the future. But that's just suggestion :)

**Comment by sjchmiela** [comment]
It's the question of whether we want to maintain our fork by `git pull --rebase`ing every time we want to update the library or whether we want to add a new preprocessor declaration to our build process. Last time we discussed, @tsapeta and I have settled on maintaining the fork. @tsapeta, what do you think now?

**Comment by mczernek** [comment]
> It's the question of whether we want to maintain our fork by `git pull --rebase`ing every time we want to update the library or whether we want to add a new preprocessor declaration to our build process. Last time we discussed, @tsapeta and I have settled on maintaining the fork. @tsapeta, what do you think now?

My suggestion is to go with declaring exact commit (tag) as a parameter of gulp task, which might be override by --commit argument. This way we still keep track of used version, prevent ourselves from accidental updates but do not need to have fork when it's not required. What do you think?

**Comment by tsapeta** [comment]
I agree that if the fork is not really required, then it would be better to just use the community repo. I don't remember why I removed HAVE_GOOGLE_MAPS flag, but maybe it is worth to check whether we still need it.

**Comment by sjchmiela** [comment]
All the Google Maps classes declarations are `ifdef` wrapped with this macro, so with this flagged undefined we don't get any react-native-maps Google Maps classes. We have to either define it or remove the macros. 🙂

**Comment by sjchmiela** [comment]
x-ref https://github.com/expo/expo-cli/pull/601

File: CHANGELOG.md [hunk 1]
```diff
@@ -41,7 +41,7 @@ This is the log of notable changes to the Expo client that are developer-facing.
 - upgraded Facebook Audience Network SDK dependency to 5.1.1 by [@sjchmiela](https://github.com/sjchmiela) ([#3394](https://github.com/expo/expo/pull/3394))
 - upgraded Facebook Core- and LoginKit dependency to 4.40.0 by [@sjchmiela](https://github.com/sjchmiela) ([#3394](https://github.com/expo/expo/pull/3394))
 - upgraded `react-native-webview` to `5.8.1` by [@sjchmiela](https://github.com/sjchmiela) ([#4146](https://github.com/expo/expo/pull/4146))
-- upgrade `react-native-maps` to `0.23.0` by [@sjchmiela](https://github.com/sjchmiela) ([#3389](https://github.com/expo/expo/pull/3389))
+- upgrade `react-native-maps` to `0.23.0` by [@sjchmiela](https://github.com/sjchmiela) ([#3389](https://github.com/expo/expo/pull/3389), [#4158](https://github.com/expo/expo/pull/4158))
```
Comment (tsapeta) [inline CHANGELOG.md]:
```suggestion
- upgrade `react-native-maps` to `0.24.2` by [@sjchmiela](https://github.com/sjchmiela) ([#3389](https://github.com/expo/expo/pull/3389), [#4158](https://github.com/expo/expo/pull/4158))
```
❓

**Review by tsapeta — APPROVED** [review]

**Pull request merged**